aeson 2.0 support

When trying to compile matrix-client with aeson 2.0 I get the following error. For compatibility with stackage-nightly this should be fixed when convenient. Thank you!
```
[3 of 7] Compiling Network.Matrix.Identity

src/Network/Matrix/Identity.hs:126:82: error:
    • Couldn't match type: Data.Aeson.KeyMap.KeyMap Value
                     with: HM.HashMap Text Value
      Expected: HM.HashMap Text Value
        Actual: aeson-2.0.3.0:Data.Aeson.Types.Internal.Object
    • In the first argument of ‘HM.toList’, namely ‘kv’
      In the second argument of ‘mapMaybe’, namely ‘(HM.toList kv)’
      In the second argument of ‘($)’, namely
        ‘mapMaybe toTuple (HM.toList kv)’
    |
126 |       (Object kv) -> pure . IdentityLookupResponse $ mapMaybe toTuple (HM.toList kv)
    |                                                                                  ^^
```

--- a/matrix-client/CHANGELOG.md
+++ b/matrix-client/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Changelog
 
+## 0.1.4.1
+
+- Support aeson-2.0
+
 ## 0.1.4.0
 
 - Completes The Room API

--- a/matrix-client/matrix-client.cabal
+++ b/matrix-client/matrix-client.cabal
@@ -31,7 +31,7 @@ source-repository head
 common common-options
   build-depends:       base                   >= 4.11.0.0 && < 5
                      , aeson-casing           >= 0.2.0.0  && < 0.3.0.0
-                     , aeson                  >= 1.0.0.0  && < 1.6
+                     , aeson                  >= 1.0.0.0  && < 2.1
   ghc-options:         -Wall
                        -Wcompat
                        -Widentities

--- a/matrix-client/src/Network/Matrix/Identity.hs
+++ b/matrix-client/src/Network/Matrix/Identity.hs
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE RecordWildCards #-}
 
@@ -40,6 +41,9 @@ import Data.Aeson (FromJSON (..), Value (Object, String), encode, object, (.:), 
 import Data.ByteString.Lazy (fromStrict)
 import Data.ByteString.Lazy.Base64.URL (encodeBase64Unpadded)
 import Data.Digest.Pure.SHA (bytestringDigest, sha256)
+#if MIN_VERSION_aeson(2,0,0)
+import qualified Data.Aeson.KeyMap as KeyMap
+#endif
 import qualified Data.HashMap.Strict as HM
 import Data.List (lookup)
 import Data.List.NonEmpty (NonEmpty)
@@ -49,6 +53,14 @@ import Data.Text.Encoding (encodeUtf8)
 import Data.Text.Lazy (toStrict)
 import qualified Network.HTTP.Client as HTTP
 import Network.Matrix.Internal
+
+#if MIN_VERSION_aeson(2,0,0)
+toKVList :: KeyMap.KeyMap v -> [(Text, v)]
+toKVList = HM.toList . KeyMap.toHashMapText
+#else
+toKVList :: HM.HashMap Text v -> [(Text, v)]
+toKVList = HM.toList
+#endif
 
 -- $setup
 -- >>> import Data.Aeson (decode)
@@ -123,7 +135,7 @@ instance FromJSON IdentityLookupResponse where
   parseJSON (Object v) = do
     mappings <- v .: "mappings"
     case mappings of
-      (Object kv) -> pure . IdentityLookupResponse $ mapMaybe toTuple (HM.toList kv)
+      (Object kv) -> pure . IdentityLookupResponse $ mapMaybe toTuple (toKVList kv)
       _ -> mzero
     where
       toTuple (k, String s) = Just (HashedAddress k, UserID s)

